### PR TITLE
SHOT-3496: VRED Refactor

### PR DESCRIPTION
**Pull request by staceyoue**
- [SHOT-3388] Remove operations.py and move methods into engine and/or hook
- [SHOT-3505] Refactor Shotgun menu generation in VRED
- [SHOT-3373] Remove publish plugin for geometry and related methods
- [SHOT-3501] Hide "Jump to file system" when the context has no filesystem

### CODE REVIEW

File: python/tk_vred/__init__.py [hunk 1]
```diff
@@ -8,5 +8,4 @@
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
-from .menu import VREDMenu
-from .operations import VREDOperations
+from .menu_generation import VREDMenuGenerator
```
[the remark this answers is not in the document]
Comment (staceyoue) [reply python/tk_vred/__init__.py:11]: @rob-aitchison -- is there a way to define this without Hound complaining about it, or is this OK?

[the remark this answers is not in the document]
Comment (rob-aitchison) [reply python/tk_vred/__init__.py:11]: Hey @staceyoue , it looks like the combination of .hound.yml and .flake8 files can fix this; I looked at https://github.com/shotgunsoftware/tk-core to deduce this. :)

[the remark this answers is not in the document]
Comment (staceyoue) [reply python/tk_vred/__init__.py:11]: This seems to have resolved the issue -- thanks!

**Review by rob-aitchison — DISMISSED** [review]
Looks great, thanks!

File: hooks/tk-multi-workfiles2/basic/scene_operation.py [hunk 2]
```diff
@@ -14,39 +22,62 @@ def execute(
...
+                                all others     - True
+        """
 
         engine = self.parent.engine
-        operations = engine.operations
-        logger = engine.logger
 
-        if operation == "current_path":
-            current_path = operations.get_current_file()
+        engine.logger.debug(
```
Comment (barbara-darkshot) [inline hooks/tk-multi-workfiles2/basic/scene_operation.py:50]: same here for the hook logger instead of the engine's

**Review by barbara-darkshot — CHANGES_REQUESTED** [review]
hey @staceyoue , I've attached some comments to the code review. Some of them are little changes and others are more thoughts :-)

**Review by barbara-darkshot — APPROVED** [review]
LGTM!

**Review by rob-aitchison — APPROVED** [review]
Looks good!

**Pull request merged**